### PR TITLE
[BOOST-4877] feat support max rewards in variable criteria incentives

### PR DESCRIPTION
**Pull request by Quazia**
🚨 Please review the [guidelines for contributing](https://github.com/boostxyz/boost-protocol/blob/main/.github/CONTRIBUTING.md) to this repository.

### Description
We enforce on chain a maximum reward per chain. Also includes protocol and SDK test updates.

💔 Thank you!

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a85c120fbb75e6c8b71fa44d89e2c5bdd9ec3c7a
</p>

File: packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts [hunk 1]
```diff
@@ -319,13 +351,15 @@ export function gasRebateIncentiveCriteria(): IncentiveCriteria {
...
  * @param {bigint} param0.limit - The total limit of the asset distribution.
+ * @param {bigint} param0.maxReward - The maximum value claimable from a single completion.
  * @param {IncentiveCriteria} param0.criteria - The incentive criteria for reward distribution.
  * @returns {Hex}
  */
 export function prepareERC20VariableCriteriaIncentivePayload({
   asset,
   reward,
   limit,
+  maxReward = 0n,
```
Comment (jonathandiep) [inline packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts:362]: Should we default the `maxReward` to the value of `reward`?

Comment (Quazia) [reply packages/sdk/src/Incentives/ERC20VariableCriteriaIncentive.ts:362]: [If maxReward is zero it'll be ignored](https://github.com/boostxyz/boost-protocol/pull/245/files#diff-d49612957cd995a8ab9ca7131b0c81c7b906f6e9b59c3343e6d23164218ff628R67)

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6409658e2e22accaa8f2f8c24cf8dddd99e551f4
</p>

**Review by sammccord — APPROVED** [review]

**Review by jonathandiep — APPROVED** [review]

**Pull request merged**